test/functional/shada/merging_spec.lua

```diff
@@ -13,6 +13,11 @@ local read_shada_file = shada_helpers.read_shada_file
 local wshada, sdrcmd, shada_fname =
   get_shada_rw('Xtest-functional-shada-merging.shada')
 
+local mock_file_path = '/a/b/'
+if helpers.iswin() then
+  mock_file_path = 'C:\\a\\b\\'
+end
```

Comment:
I'm not sure this is the right approach. If a path with `/` slashes gets stored in shada, nvim should still be able to understand it. @ZyX-I what do you think?

Reply:
I agree. However, I was trying to construct a payload with a filename, that has more than 6 characters in the path and as far as I understand the payload part with the filename should be something like `\xc4\x07/a/bb/c` but that does not work, however `\xc4\x05/aa/c` does work, which leaves me completely baffled. I was reading on the [bin format in msgpack spec](https://github.com/msgpack/msgpack/blob/master/spec.md#bin-format-family).

The example test file:
```lua
-- ShaDa filename length greater that 6 test
local helpers = require('test.functional.helpers')(after_each)
local exc_exec = helpers.exc_exec
local eq = helpers.eq
local shada_helpers = require('test.functional.shada.helpers')
local reset, clear, get_shada_rw =
  shada_helpers.reset, shada_helpers.clear, shada_helpers.get_shada_rw
local wshada, sdrcmd, shada_fname =
  get_shada_rw('Xtest-functional-shada-merging.shada')

describe('When trying to par ShaDa code this', function()
  before_each(reset)
  after_each(function()
    clear()
    os.remove(shada_fname)
  end)

  it('works #7287',
  function()
    wshada('\007\001\018\131\162mX\195\161f\xc4\006/a/b/-\161nA')
    eq(0, exc_exec(sdrcmd()))
  end)

  it('does not work #7287',
  function()
    wshada('\007\001\018\131\162mX\195\161f\xc4\007/a/bc/-\161nA')
    eq(0, exc_exec(sdrcmd()))
  end)
end)
```

Do you know if there is any documentation on the protocol that NVIM uses in order to store things?